recipe/meta.yaml

```
@@ -15,11 +15,13 @@ source:
     - setup_cmake_args.patch
     - fix_libtools_error.patch  # [osx]
     - unvendor_pybind11.patch
-    - do_not_try_cuda_test_compile.patch  # [linux and aarch64]
+    - use_new_cudatoolkit.patch  # [cuda_compiler_version != "None"]
+    - allow_any_generator.patch
 
 build:
-  number: 4
-  skip: true  # [aarch64 and cuda_compiler_version != "None"]
+  number: 5
+  skip: true  # [aarch64 and cuda_compiler_version == "12.0"]
```

Added a tracking issue. Maybe it is worth linking here?

Also added a suggestion to make the CUDA 12 check more robust across minor versions

```suggestion
  # xref: https://github.com/conda-forge/dlib-feedstock/issues/84
  skip: true  # [aarch64 and (cuda_compiler_version or "").startswith("12")]
```